src/com/facebook/buck/android/SmartDexingStep.java

```diff
@@ -463,9 +472,20 @@ static void createDxStepForDxPseudoRule(
       OptionalInt xzCompressionLevel,
       Optional<String> dxMaxHeapSize,
       String dexTool,
-      @Nullable Collection<Path> classpathFiles) {
+      @Nullable Collection<Path> classpathFiles,
+      boolean useDexBuckedId) {
 
+    Optional<String> buckedId = Optional.empty();
     String output = outputPath.toString();
+    if (useDexBuckedId) {
+      // We know what the output file name is ("classes.dex" or "classesN.dex") as these
+      // are generated in SplitZipStep and passed around as part of a multi-map - it is
+      // simply easier and cleaner to extract the dex file number to be used as unique
+      // identifier rather than creating another map and pass it around
+      String[] tokens = Files.getNameWithoutExtension(outputPath.toString()).split("classes");
+      String id = tokens.length == 1 ? "" /* primary */ : tokens[1] /* secondary */;
```

Comment:
This needs to have "tokens.length == 0", the same as it was in the original change.

Reply:
interesting. ok will make the change to see if the tests pass finally :)

Reply:
Done and pushed. Lets wait for CI :)